File: C#/relatorio-consolidado.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

class Program
{
    // Definição do endpoint da API
    private static string url = "https://api.mailgrid.net.br/consolidate/";

    static async Task Main(string[] args)
    {
        // Dados de autenticação e parâmetros de consulta
        var data = new
        {
            usuario_smtp = "smtp1@seudominio.com.br", // Usuário SMTP para autenticação
            senha_smtp = "xxxxxxxx", // Senha SMTP para autenticação
            dataini = "2023-02-01", // Data de início da consulta (AAAA-MM-DD)
            horaini = "00:01", // Hora de início da consulta (HH:MM)
            datafim = "2023-08-30", // Data de fim da consulta (AAAA-MM-DD)
            horafim = "23:59" // Hora de fim da consulta (HH:MM)
        };

        // Inicializa o HttpClient
        using (var client = new HttpClient())
        {
            // Configura o cabeçalho da requisição para JSON
            client.DefaultRequestHeaders.Add("Content-Type", "application/json");

            // Serializa os dados para JSON
            var jsonData = JsonConvert.SerializeObject(data);
            var content = new StringContent(jsonData, Encoding.UTF8, "application/json");

            try
            {
                // Envia a requisição POST
                HttpResponseMessage response = await client.PostAsync(url, content);

                // Obtém o código de status HTTP
                int statusCode = (int)response.StatusCode;
                Console.WriteLine($"Código de status: {statusCode}");

                // Verifica se a requisição foi bem-sucedida (código 2xx)
                if (response.IsSuccessStatusCode)
                {
                    // Lê a resposta como string
                    string responseBody = await response.Content.ReadAsStringAsync();

                    // Tenta desserializar a resposta JSON
                    var decodedResponse = JsonConvert.DeserializeObject(responseBody);

                    // Exibe a resposta formatada em JSON
                    Console.WriteLine("Resposta da API:");
                    string formattedResponse = JsonConvert.SerializeObject(decodedResponse, Formatting.Indented);
                    Console.WriteLine(formattedResponse);
                }
                else
                {
                    // Exibe erro caso o código de status seja maior ou igual a 400
                    Console.WriteLine($"Erro na requisição. Código de status HTTP: {statusCode}");
                }
            }
            catch (Exception ex)
            {
                // Exibe qualquer erro durante a requisição
                Console.WriteLine($"Erro ao processar a requisição: {ex.Message}");
            }
        }
    }
}

File: java/relatorio-consolidado.java
import java.io.*;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class mailgridConsolidate {
	public static void main(String[] args) {
		// Definindo o endpoint da API
		String url = "https://api.mailgrid.net.br/consolidate/";

		// Dados de autenticação e parâmetros de consulta
		String token_auth = "INFORME-O-TOKEN-AQUI"; // Token de autenticação
		String dataInicio = "2023-02-01"; // Data de início da consulta (AAAA-MM-DD)
		String horaInicio = "00:01"; // Hora de início da consulta (HH:MM)
		String dataFim = "2023-08-30"; // Data de fim da consulta (AAAA-MM-DD)
		String horaFim = "23:59"; // Hora de fim da consulta (HH:MM)

		// Criando o JSON de entrada
		String jsonInputString = "{"
				+ "\"token_auth\": \"" + token_auth + "\","
				+ "\"dataini\": \"" + dataInicio + "\","
				+ "\"horaini\": \"" + horaInicio + "\","
				+ "\"datafim\": \"" + dataFim + "\","
				+ "\"horafim\": \"" + horaFim + "\""
				+ "}";

		try {
			// Criando a URL e a conexão HTTP
			URL urlObj = new URL(url);
			HttpURLConnection connection = (HttpURLConnection) urlObj.openConnection();
			connection.setRequestMethod("POST");

			// Definindo os cabeçalhos da requisição
			connection.setRequestProperty("Content-Type", "application/json");
			connection.setDoOutput(true); // Permite que dados sejam enviados no corpo da requisição

			// Enviando os dados JSON no corpo da requisição
			try (OutputStream os = connection.getOutputStream()) {
				byte[] input = jsonInputString.getBytes(StandardCharsets.UTF_8);
				os.write(input, 0, input.length); // Escreve os dados no corpo da requisição
			}

			// Obtendo o código de status da resposta
			int statusCode = connection.getResponseCode();

			// Lendo a resposta da API
			try (BufferedReader br = 
			new BufferedReader(new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8))) {
				StringBuilder response = new StringBuilder();
				String line;
				while ((line = br.readLine()) != null) {
					response.append(line);
				}

				// Verificando o código de status e exibindo a resposta
				if (statusCode == HttpURLConnection.HTTP_OK) {
					// Se o código de status for 200 (OK), exibe a resposta formatada
					System.out.println("Resposta da API: ");
					System.out.println(response.toString());
				} else {
					// Se o código de status for diferente de 200, exibe o erro
					System.out.println("Erro na requisição. Código de status HTTP: " + statusCode);
				}
			}
		} catch (IOException e) {
			// Caso ocorra um erro durante a requisição, exibe a mensagem de erro
			System.out.println("Erro na requisição: " + e.getMessage());
		}
	}
}
